### PR TITLE
Added "inplace" keyword argument to "Unit.convert"

### PR DESCRIPTION
**Pull request by bekozi**
- Changed "convert" to in-place for float-to-float conversions.
- Revised "convert" docstring.
- Updated unit test.

### CODE REVIEW

File: cf_units/__init__.py [hunk 1]
```diff
@@ -1849,10 +1849,10 @@ def __ne__(self, other):
         """
         return not self == other
 
-    def convert(self, value, other, ctype=FLOAT64):
+    def convert(self, value, other, ctype=FLOAT64, inplace=True):
```
Comment (pelson) [inline cf_units/__init__.py]: I'm not convinced this is a sensible default. I suspect the majority of users will want to _not_ modify the data they are converting in-place.

Comment (ajdawson) [inline cf_units/__init__.py]: I agree. It also changes the current behaviour without warning. If this is `False` by default then I think it is OK.

**Comment by bekozi** [comment]
Got it. Changed the default to `False`.

File: cf_units/__init__.py [hunk 2]
```diff
@@ -1883,27 +1887,26 @@ def convert(self, value, other, ctype=FLOAT64):
             32.0
             >>> a64 = np.arange(10, dtype=np.float64)
             >>> c.convert(a64, f)
-            array([ 32. ,  33.8,  35.6,  37.4,  39.2,  41. ,  42.8,  44.6, \
- 46.4,  48.2])
+            array([ 32. ,  33.8,  35.6,  37.4,  39.2,  41. ,  42.8,  44.6,
+                    46.4,  48.2])
```
Comment (pelson) [inline cf_units/__init__.py]: Hmmm, I wonder if this would break the doctest now. Either way, it looks better! :+1:

**Comment by pelson** [comment]
:+1: from me. @ajdawson - would you mind casting your eyes on it again and confirming you are happy too?

File: cf_units/__init__.py [hunk 3]
```diff
@@ -1849,10 +1849,10 @@ def __ne__(self, other):
         """
         return not self == other
 
-    def convert(self, value, other, ctype=FLOAT64):
+    def convert(self, value, other, ctype=FLOAT64, inplace=False):
         """
-        Converts a single value or numpy array of values from the current unit
-        to the other target unit.
+        Converts a single ``value`` or NumPy array of ``value``s from the
```
Comment (ajdawson) [inline cf_units/__init__.py]: I thought we used double back-ticks to denote actualt variables or values like `False` or `None`, I'm not sure they are appropriate here as the word value is generic and not referring to the input argument `value`.

File: cf_units/__init__.py [hunk 4]
```diff
@@ -1867,9 +1867,13 @@ def convert(self, value, other, ctype=FLOAT64):
...
             when `value` is not a NumPy array or is a NumPy array composed of
             NumPy integers. The default is 64-bit double-precision conversion.
+        * inplace (bool):
+            If ``False``, return a deep copy of the value array. If ``True``,
+            convert the values in-place. A new array will be created if
+            ``value`` is an integer NumPy array.
 
         Returns:
-            float or numpy.ndarray of appropriate float type.
+            float or numpy.ndarray of appropriate float type
```
Comment (ajdawson) [inline cf_units/__init__.py]: Why did you get rid of the full stop? If you think it reads better you could change it to this:

```
A float or ``numpy.ndarray`` of appropriate float type.
```

File: cf_units/__init__.py (hunk 2 above)
Comment (ajdawson) [inline cf_units/__init__.py]: Does this have a doctest running on it? It certainly looks nicer this way, and if it doesn't break anything I prefer it this way.

File: cf_units/__init__.py [hunk 5]
```diff
@@ -1883,27 +1887,26 @@ def convert(self, value, other, ctype=FLOAT64):
...
             array([ 32.        ,  33.79999924,  35.59999847,  37.40000153,
                     39.20000076,  41.        ,  42.79999924,  44.59999847,
                     46.40000153,  48.20000076], dtype=float32)
 
-        .. note::
-
-            Conversion is done *in-place* for numpy arrays. Also note that,
-            conversion between unit calendars is not permitted.
-
+        .. note:: Conversion between unit calendars is not permitted.
```
Comment (ajdawson) [inline cf_units/__init__.py]: I think this should still be split over multiple lines like it was previously:

```
.. note::

   Conversion between unit calendars is not permitted.
```

There should also be a blank line after the note before the end of the comment.

**Comment by ajdawson** [comment]
I made a few comments, all minor. @pelson - I'm not too familiar with this code (or ctypes in general) but it looks reasonable enough.

File: cf_units/__init__.py (hunk 2 above)
Comment (bekozi) [inline cf_units/__init__.py]: They aren't running using `python setup.py test`. I did a `py.test --doctest-modules` and got some failures including this one. The line break drove me crazy, so I reduced the test arrays to size 3 to fit on a single line. There are three other failures, but I don't think I should fix those (they look minor).

File: cf_units/__init__.py (hunk 3 above)
Comment (bekozi) [inline cf_units/__init__.py]: Too many values. You're right, no ticks should be used here.

File: cf_units/__init__.py (hunk 4 above)
Comment (bekozi) [inline cf_units/__init__.py]: I'll add it back. I don't use them in docstrings like this, so I removed it out of habit...

File: cf_units/__init__.py (hunk 5 above)
Comment (bekozi) [inline cf_units/__init__.py]: Done.

**Pull request merged**